core/src/test/java/io/questdb/std/datetime/microtime/TimestampsTest.java

```diff
@@ -151,6 +151,13 @@ public void testCeilYYYY() throws Exception {
         TestUtils.assertEquals("2009-01-01T00:00:00.000Z", sink);
     }
 
+    @Test
+    public void testCeilWW() throws Exception {
+        long micros = TimestampFormatUtils.parseTimestamp("2024-01-02T23:59:59.999Z");
+        TimestampFormatUtils.appendDateTime(sink, Timestamps.ceilWW(micros));
+        TestUtils.assertEquals("2024-01-08T00:00:00.000Z", sink);
+    }
+
     @Test
```

Comment:
ISO Week parsing is buggy, this tests should pass

```
    @Test
    public void testParseWW() throws NumericException {
        DateFormat byWeek = getPartitionDirFormatMethod(PartitionBy.WEEK);
        try {
            byWeek.parse("2020W00", null);
            Assert.fail("ISO Week 00 is invalid");
        } catch (NumericException ignore) {
        }

        try {
            byWeek.parse("2020W54", null);
            Assert.fail();
        } catch (NumericException ignore) {
        }

        Assert.assertEquals("2019-12-30T00:00:00.000Z", Timestamps.toString(byWeek.parse("2020W01", null)));
        Assert.assertEquals("2020-12-28T00:00:00.000Z", Timestamps.toString(byWeek.parse("2020W53", null)));
        Assert.assertEquals("2021-01-04T00:00:00.000Z", Timestamps.toString(byWeek.parse("2021W01", null)));

        try {
            byWeek.parse("2019W53", null);
            Assert.fail("2019 has 52 ISO weeks");
        } catch (NumericException ignore) {
        }

        Assert.assertEquals("2019-12-30T00:00:00.000Z", Timestamps.toString(byWeek.parse("2020W01", null)));
        Assert.assertEquals("2014-12-29T00:00:00.000Z", Timestamps.toString(byWeek.parse("2014W53", null)));
        Assert.assertEquals("2014-12-29T00:00:00.000Z", Timestamps.toString(byWeek.parse("2014W53", null)));
    }
```

At the moment it fails because it parses `2020W00` and `2019W53`